pymc/step_methods/metropolis.py

```diff
@@ -67,38 +67,29 @@ def __init__(self, s):
 
 class NormalProposal(Proposal):
     def __call__(self, rng: Optional[np.random.Generator] = None):
-        if rng is None:
-            rng = nr
-        return rng.normal(scale=self.s)
+        return (rng or nr).normal(scale=self.s)
```

Comment:
These changes resolve two `mypy` errors:
1. `rng` is no longer reassigned, and 
2. the `(rng or nr)` has type `Generator | Module("numpy.random")` which resolves `Item "None" of "Optional[Generator]" has no attribute "normal"`

Reply:
Looks much less readable though